stix2validator/shoulds.py

```diff
@@ -274,14 +274,16 @@ def relationships_strict(instance):
         return
 
     r_type = instance['relationship_type']
-    r_source = re.search("(.+)\-\-", instance['source_ref']).group(1)
-    r_target = re.search("(.+)\-\-", instance['target_ref']).group(1)
+    try:
+        r_source = re.search("(.+)\-\-", instance['source_ref']).group(1)
+        r_target = re.search("(.+)\-\-", instance['target_ref']).group(1)
+    except (AttributeError, TypeError):
+        return  # Schemas already catch these
 
     if (r_type in enums.COMMON_RELATIONSHIPS or
             r_source in enums.NON_SDOS or
             r_target in enums.NON_SDOS):
-        # Schemas will already catch relationships not allowed by spec
-        return
+        return  # Schemas already catch invalid relationships
```

Comment:
In response to the first two questions: yes to both. The schema validation will give an error if the type of object in the source or target cannot have a relationship to/from it: https://github.com/clenk/cti-stix2-json-schemas/blob/64d8a68930f78886e7e59bc085522b6922fdfdc2/schemas/sros/relationship.json#L41 If that's the case, or if the relationship type is one that all objects can have, then we don't need to do the following checks, which ensure that the relationship type is listed in the spec for the given source type and target type.